### PR TITLE
Clean up old `GFileCache` implementation

### PR DESCRIPTION
**Pull request by ayaka14732**
This PR is a follow-up of https://github.com/google/jax/pull/21926. It is part of the efforts to add LRU cache eviction policy for the JAX persistent compilation cache.

Since we now have the new `LRUChache` class, which supports both local filesystem and GCS, the old `GFileCache` class can be removed.

### CODE REVIEW

**Comment by ayaka14732** [comment]
Manually tested on GPU VM and it works.

File: jax/_src/path.py [hunk 1]
```diff
@@ -19,8 +19,10 @@
 
 try:
   import etils.epath as epath
+  epath_installed = True
```
Comment (superbobry) [inline jax/_src/path.py:22]: I guess you could use `epath is not None` instead of `epath_installed`?

File: jax/_src/lru_cache.py [hunk 2]
```diff
@@ -56,29 +53,26 @@ def __init__(self, path: str, *, max_size: int, lock_timeout_secs: float | None
         indicates no limit, allowing the cache size to grow indefinitely.
       lock_timeout_secs: (optional) The timeout for acquiring a file lock.
     """
-    # TODO(ayx): add support for cloud other filesystems such as GCS
-    if not self._is_local_filesystem(path):
-      raise NotImplementedError("LRUCache only supports local filesystem at this time.")
+    if not self._is_local_filesystem(path) and not pathlib.epath_installed:
```
Comment (superbobry) [inline jax/_src/lru_cache.py:56]: Nit: make this staticmethod a function.

**Review by superbobry — APPROVED** [review]

File: jax/_src/lru_cache.py (hunk 2 above)
Comment (ayaka14732) [reply jax/_src/lru_cache.py]: Done

File: jax/_src/path.py (hunk 1 above)
Comment (ayaka14732) [reply jax/_src/path.py:22]: I think having a `epath_installed` is clearer, because it is imported in another file and used multiple times.

**Pull request merged**